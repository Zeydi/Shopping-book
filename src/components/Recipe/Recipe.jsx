import React from "react";

const Recipe = props => {
  return (
    <div className="container">
      <div className="collection">
        <li className="collection-item">
          <b>Total: {props.totalPrice} $</b>
        </li>
        <li className="collection-item">
          <b>Discount: {props.bestOffer} $</b>
        </li>
        <li className="collection-item">
          <b>To pay: {props.totalPrice - props.bestOffer} $</b>
        </li>
      </div>
    </div>
  );
};

export default Recipe;
