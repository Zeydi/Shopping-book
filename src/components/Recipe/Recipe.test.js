import React from "react";
import { shallow } from "enzyme";
import Recipe from "./Recipe";

describe("Recipe Component", () => {
  const totalPrice = 10;
  const bestOffer = 5;

  const recipe = shallow(
    <Recipe totalPrice={totalPrice} bestOffer={bestOffer} />
  );

  it("renders correctly", () => {
    expect(recipe).toMatchSnapshot();
  });

  it("shows all received props", () => {
    expect(recipe.contains(<b>Total: {totalPrice} $</b>)).toBeTruthy();
    expect(recipe.contains(<b>Discount: {bestOffer} $</b>)).toBeTruthy();
    expect(
      recipe.contains(<b>To pay: {totalPrice - bestOffer} $</b>)
    ).toBeTruthy();
  });
});
