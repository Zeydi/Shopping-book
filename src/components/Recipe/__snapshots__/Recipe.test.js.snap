// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Component renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="collection"
  >
    <li
      className="collection-item"
    >
      <b>
        Total: 
        10
         $
      </b>
    </li>
    <li
      className="collection-item"
    >
      <b>
        Discount: 
        5
         $
      </b>
    </li>
    <li
      className="collection-item"
    >
      <b>
        To pay: 
        5
         $
      </b>
    </li>
  </div>
</div>
`;
